domain/secret/watcher_test.go

```diff
@@ -348,51 +364,68 @@ func (s *watcherSuite) TestWatchObsoleteUserSecretsToPrune(c *gc.C) {
 	uri2 := coresecrets.NewURI()
 	c.Logf("uri1: %v, uri2: %v", uri1, uri2)
 
-	err := createUserSecret(ctx, st, 1, uri1, secret.UpsertSecretParams{
-		Data:       data,
-		RevisionID: ptr(uuid.MustNewUUID().String()),
-	})
-	c.Assert(err, jc.ErrorIsNil)
-	err = createUserSecret(ctx, st, 1, uri2, secret.UpsertSecretParams{
-		Data:       data,
-		AutoPrune:  ptr(true),
-		RevisionID: ptr(uuid.MustNewUUID().String()),
-	})
-	c.Assert(err, jc.ErrorIsNil)
-
 	w, err := svc.WatchObsoleteUserSecretsToPrune(ctx)
 	c.Assert(err, gc.IsNil)
 	c.Assert(w, gc.NotNil)
 	defer workertest.CleanKill(c, w)
 
-	wc := watchertest.NewNotifyWatcherC(c, w)
-
-	// Wait for the initial changes.
-	wc.AssertOneChange()
+	harness := watchertest.NewHarness(s, watchertest.NewWatcherC(c, w))
+	harness.AddTest(func(c *gc.C) {
+		err := createUserSecret(ctx, st, 1, uri1, secret.UpsertSecretParams{
+			Data:       data,
+			RevisionID: ptr(uuid.MustNewUUID().String()),
+		})
+		c.Assert(err, jc.ErrorIsNil)
+		err = createUserSecret(ctx, st, 1, uri2, secret.UpsertSecretParams{
+			Data:       data,
+			AutoPrune:  ptr(true),
+			RevisionID: ptr(uuid.MustNewUUID().String()),
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	}, func(w watchertest.WatcherC[struct{}]) {
+		w.AssertNoChange()
+	})
 
-	// create revision 2, no event is fired because the auto prune is not turned on for uri1.
-	createNewRevision(c, st, uri1)
-	wc.AssertNoChange()
+	harness.AddTest(func(c *gc.C) {
+		// create revision 2, no event is fired because the auto prune is not turned on for uri1.
+		createNewRevision(c, st, uri1)
+	}, func(w watchertest.WatcherC[struct{}]) {
+		w.AssertNoChange()
+	})
 
-	// create revision 2, and obsolete revision 1. An event is fired because the auto prune is turned on for uri2.
-	createNewRevision(c, st, uri2)
-	wc.AssertNChanges(2)
```

Comment:
Can we port this to `WatcherC[T]`
Will be useful and we can avoid the multiple `AssertChange()` calls below.